.vsts.pipelines/jobs/collect-artifacts.yml

```diff
@@ -0,0 +1,104 @@
+jobs:
+- job: Collect_Artifacts
+
+  dependsOn: ${{ parameters.dependsOn }}
+
+  displayName: Collect Artifacts
+
+  pool: ${{ parameters.pool }}
+
+  steps:
+  - ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
+    - task: DownloadBuildArtifacts@0
+      displayName: Download CentOS artifacts
+      inputs:
+        artifactName: Tarball centos71 Offline
+        downloadPath: '$(Build.StagingDirectory)/centos'
+    - task: DownloadBuildArtifacts@0
+      displayName: Download Debian artifacts
+      inputs:
+        artifactName: Tarball debian9 Offline
+        downloadPath: '$(Build.StagingDirectory)/debian'
+    - task: DownloadBuildArtifacts@0
+      displayName: Download Fedora artifacts
+      inputs:
+        artifactName: Tarball fedora29 Offline
+        downloadPath: '$(Build.StagingDirectory)/fedora'
+    - task: DownloadBuildArtifacts@0
+      displayName: Download Ubuntu artifacts
+      inputs:
+        artifactName: Tarball ubuntu1604 Offline
+        downloadPath: '$(Build.StagingDirectory)/ubuntu'
+    - task: DownloadBuildArtifacts@0
+      displayName: Download portable artifacts
+      inputs:
+        artifactName: Tarball centos71 Offline Portable
+        downloadPath: '$(Build.StagingDirectory)/portable'
+
+    - task: Bash@3
+      inputs:
+        targetType: 'inline'
+        workingDirectory: '$(Build.StagingDirectory)'
+        script: |
+          version=`echo centos/*/Private.SourceBuilt.Artifacts*.tar.gz | sed 's/\.tar\.gz$//' | sed 's/^centos\/.*\/Private\.SourceBuilt\.Artifacts\.//'`
+          echo "Artifacts version: $version"
+          mkdir collected
+          cd collected
+          # do CentOS last to prefer those versions of any duplicates
+          for os in portable debian fedora ubuntu centos; do
+            # DownloadArtifact creates another directory - use '*' to not deal with this
+            tar xzf ../$os/*/Private.SourceBuilt.Artifacts*.tar.gz
+            rm -rf ../$os
+          done
+          echo $(Build.SourceVersion) > .version
+          echo $(Build.BuildNumber) > .buildnumber
+          echo $(Build.BuildId) > .buildid
+          tar czf ../Private.SourceBuilt.Artifacts.$version.tar.gz * .version .buildnumber .buildid
+          rm -rf *
+          mv ../Private.SourceBuilt.Artifacts.*.tar.gz .
```

Comment:
It isn't clear to me what this artifact will be used for... sounds like more than diag if it's ending in BAR, so I'm a little concerned since point of source-build is that you can't get a combined artifact like this.